File: php/ltx/src/InterplanetLTX/NodeUrl.php
<?php
/**
 * NodeUrl.php — Per-node perspective URL
 * Story 33.4 — PHP LTX library
 */

namespace InterplanetLTX;

/** A per-node perspective URL for an LTX session. */
readonly class NodeUrl
{
    public function __construct(
        public string $nodeId,
        public string $name,
        public string $role,
        public string $url,
    ) {}
}

File: java/ltx/src/com/interplanet/ltx/NodeUrl.java
package com.interplanet.ltx;

/**
 * NodeUrl — a perspective URL for a specific node in an LTX session.
 * Story 33.2 — Java LTX library
 */
public record NodeUrl(
    /** Node identifier, e.g. "N0". */
    String nodeId,
    /** Node display name. */
    String name,
    /** Node role: "HOST" or "PARTICIPANT". */
    String role,
    /** Full URL including ?node= and #l= parameters. */
    String url
) {}
